### ISSUE TITLE
Play order plugins should handle song list changes

### ISSUE THREAD

**Issue by lazka** [quodlibet/quodlibet#575]
Original [issue 575](https://code.google.com/p/quodlibet/issues/detail?id=575) created by reiter.christoph on 2010-09-26T23:18:27.000Z:

Play order plugins currently reset if the (sort) order changes or songs get removed/added.

This makes it possible for a song to get played twice in &quot;shuffle&quot; mode if a song from the song list gets removed. Also the song history gets lost and &quot;previous&quot; stops the player.

**Comment by Paveloom** [comment]
Yup, still annoying. Just got a song repeated twice in a row (excluding the playing one that I've deleted from the playlist) in a 300+ songs playlist.
